SQOOP-3423 Let user pass password to connect Hive when it set to LDAP…

dionusos opened the pull request:
… authentication

Commit:
SQOOP-3423 Let user pass password to connect Hive when it set to LDAP authentication

Change-Id: I8a8e9abf092195405365a789c93a2987be5e9304

szvasas commented on src/java/org/apache/sqoop/hive/HiveServer2ConnectionFactoryInitializer.java:
```diff
@@ -53,6 +57,13 @@ private String determineConnectionUsername(SqoopOptions sqoopOptions) {
     }
   }
 
+  private String determineConnectionPassword(SqoopOptions sqoopOptions) {
```
Is the empty string ("") a legal password in LDAP environment?
If yes then we could eliminate this method and use sqoopOptions.getHs2Password() instead.

dionusos replied:
As I found in LDAP environment empty password is possible so as the recommendation I removed the extra method.

szvasas reviewed (changes requested):
Hi @dionusos,

Thank you for the patch, it looks great, I have found only 2 minor things which need to be addressed before merging:
- hive.txt has also some documentation about the HiveServer2 connection("Currently only Kerberos authentication and text file format is supported with this option."), please make sure that it mentions the LDAP authentication too (

Commit:
SQOOP-3423 Let user pass password to connect Hive when it set to LDAP authentication
Amendum commit. Enabling usage of empy password and mentioning LDAP auth in hive.txt

Change-Id: I102ab6e520fcd0f0030106300c90c86f3ccfc7f9

ebogi commented:
LGTM, ship it!
